### ISSUE TITLE
WP6: Multiline Overlay reacts to viewport changes

### ISSUE THREAD

**Issue by Galanos** [image-source-control/image-source-control#177]
In **WP6 Beta**, Overlays that have multiline content (i. e. automatic soft line breaks) due to the corresponding image’s size are unexpectedly susceptible to changes to the viewport size.
Tested in Firefox 98.

How to reproduce
---

1. Use any current standard WordPress theme: 2020, 2021, or 2022 **in WP6**
2. Create image blocks with ISC metadata that are small enough to cause the Overlay to become multiline; Galleries work great
3. See the frontend
4. While watching the multiline Overlays, resize the viewport (pulling the browser window’s edges or viewing/hiding browser tools)
5. See the Overlays change their width with every change to the viewport until they become single-line

Expected behavior
---

I’d expect the Overlay to either keep its size or to dynamically adapt to the image dimensions. But this is weird behavior.

**Screenshots**

If needed, I’ll happily provide a video showcasing the behavior in Slack.

Priority
---

Low, users tend to not rearrange the viewport. If it’s easily fixed, it’d be nice to have it ready for the WP6 release.

Additional context
---

For Galleries, there is a [newly updated CSS attribute](https://make.wordpress.org/core/2022/05/03/block-editor-miscellaneous-dev-notes-for-wordpress-6-0/#gallery-block) `gap` which I suspected to be the culprit at first, but I excluded it later on—deactivating it simply causes the Gallery to rearrange and the multiline overlay to become single-line.

I’m speculating that this is due to calculated CSS `width` attributes for the image boxes or their parents but it’s hard for me to pinpoint.

Tests
---

—

**Comment by Galanos** [comment]
I found that this only affects Overlay placements “top-right” and “bottom-right” and updated the issue.

**Comment by webzunft** [comment]
The original problem does not seem related to WordPress 6.0.
The calculation of the position is already broken when the overlay is originally placed.

The calculation that adds the left margin when the overlay is positioned "bottom-left", like here:

<img width="420" alt="Bildschirmfoto 2022-05-19 um 13 02 42" src="https://user-images.githubusercontent.com/2728720/169278901-de7fb18e-a444-474a-922f-e12f26e94faf.png">

causes a negative left margin when using "bottom-right":

<img width="417" alt="Bildschirmfoto 2022-05-19 um 13 03 33" src="https://user-images.githubusercontent.com/2728720/169279044-d83b1461-9f56-4212-817f-3baf2833e0c4.png">

@Galanos do you agree that we should – for consistency – use a right margin for "bottom-right" to have a gab between right image border and overlay?

**Comment by webzunft** [comment]
We could also just add a margin inline CSS rule like `margin: 0 5px;` to any overlay to have a gab on both sides, regardless of whether the overlay is aligned left or right. When the overlay is large enough, users don’t see a difference between "bottom-left" and "bottom-right" anyway, and having a gap only on one end seems inconsequent.

**Comment by webzunft** [comment]
Dev note: we have defined the margins for overlays in https://github.com/image-source-control/image-source-control/blob/master/public/assets/js/captions.js#L77-L80, which is not changeable through a filter and is bad practice anyway.

**Comment by Galanos** [comment]
@webzunft Absolutely, I agree. I would’ve thought that all left-oriented style attributes would’ve been mirrored to right-side ones for right-oriented placements, anyway.

As in many other cases before, I really have no insight into how the JS scripts calculate the Overlay’s dimensions and placements, and can’t contribute to that. But if I were to write pure CSS for this, I would add left and right margins to the Overlay to leave a gap between Overlay edges and image edges at all times and content sizes, yes.

We’d have to test with very small images, too, in order to see how overflowing source info affects the image and its box. Again, hard to predetermine without deeper knowledge of the JS.

**Comment by webzunft** [comment]
I’ve iced this issue since I haven’t seen it in real life, yet.
